crates/cargo-util-schemas/src/core/partial_version.rs

```diff
@@ -183,5 +183,9 @@ fn is_req(value: &str) -> bool {
     let Some(first) = value.chars().next() else {
         return false;
     };
-    "<>=^~".contains(first) || value.contains('*') || value.contains(',')
+    "<>=^~".contains(first)
+        || value.contains('*')
+        || value.contains(',')
+        || value.contains('x')
+        || value.contains('X')
```

Comment:
While `PartialVersion` was made for `RustVersion`, in [SemVer spec](https://semver.org/#spec-item-9) alphanumerics is allowed in prerelease and build metadata. Should we make `PartialVersion` compliant to that?

Reply:
`PartialVersion` is needed for PkgIdSpecs which support pre-release, so we do need to support it.  I've updated it to handle that.